### PR TITLE
Upgrade to v2.0 compliance

### PR DESCRIPTION
**Pull request by lpender**
These modifications update the player to work with LKQD v2.0, in the manner described here:

https://wiki.lkqd.com/pages/viewpage.action?pageId=852444

and based on the example here

http://demo.lkqd.com/formats.html

### CODE REVIEW

**Comment by akestner** [comment]
Can I get a brief, high-level description?

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "gulp-sourcemaps": "^1.6.0",
     "gulp-uglify": "^1.5.1",
     "gulp-util": "^3.0.7",
-    "jquery": "^2.1.4",
+    "jquery": "^3.3.1",
```
Comment (lpender) [inline package.json]: Upgrade due to security vulnerability.

File: src/lkqd-videojs.js [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 var LkqdVideoJS = (function() {
   function LkqdVideoJS(player, options) {
     this.player = player;
 
     var defaults = {
       adTagUrl: '',
-      videoEl: '',
-      containerEl: '',
+      videoEl: { dispatchEvent: function() {} },
+      containerEl: { dispatchEvent: function() {} },
```
Comment (lpender) [inline src/lkqd-videojs.js]: Just added these as protections in case the elements weren't passed...

File: src/lkqd-videojs.js [hunk 3]
```diff
@@ -38,23 +43,10 @@ var LkqdVideoJS = (function() {
...
-      '&containerid=' + encodeURIComponent(this.options.containerEl.id) +
-      '&width=' + this.options.playerWidth +
-      '&height=' + this.options.playerHeight +
-      '&mode=' + this.options.viewMode +
-      '&c1=' + this.options.publisherId +
-      '&rnd=' + Math.floor(Math.random() * 100000000);
+
+    var { pid, sid } = this._adTagParams();
+
+    vpaidLoaderScript.src = `https://ad.lkqd.net/vpaid/formats.js?pid=${pid}&sid=${sid}`
```
Comment (lpender) [inline src/lkqd-videojs.js]: This now matches the docs for 2.0

File: src/lkqd-videojs.js [hunk 4]
```diff
@@ -69,14 +61,17 @@ var LkqdVideoJS = (function() {
       detail: _this
     });
 
-    this.options.videoEl.dispatchEvent(event);
+    this.options.containerEl.dispatchEvent(event);
```
Comment (lpender) [inline src/lkqd-videojs.js]: Dispatching from the containerEl is safer since the videoEl may not exist.

File: src/lkqd-videojs.js [hunk 5]
```diff
@@ -86,22 +81,61 @@ var LkqdVideoJS = (function() {
...
+          custom1: '',
+          custom2: '',
+          custom3: '',
+          pubMacros: '',
+          dfp: false,
+          lkqdId: iframeId,
+          supplyContentVideo: {
+            url: '', clickUrl: '', play: 'post', volume: 100, loop: false
+          }
+        }
```
Comment (lpender) [inline src/lkqd-videojs.js]: Copied from the demo

File: src/lkqd-videojs.js [hunk 6]
```diff
@@ -86,22 +81,61 @@ var LkqdVideoJS = (function() {
...
     });
   };
 
   LkqdVideoJS.prototype._onAdLoaded = function () {
     var _this = this;
 
+    if (_this.options.videoEl) {
+      _this.options.videoEl.style.display = "none";
+    }
+
```
Comment (lpender) [inline src/lkqd-videojs.js]: We need to hide the videoEl or they both appear and stack on top of each other.

File: src/lkqd-videojs.js (hunk 3 above)
Comment (akestner) [reply src/lkqd-videojs.js]: 👍

File: src/lkqd-videojs.js (hunk 5 above)
Comment (akestner) [reply src/lkqd-videojs.js]: 👍

**Review by akestner — APPROVED** [review]
Generally looks good, couple questions around defaults, but nothing that should prevent this from landing.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (lpender) [reply package.json]: Yes

File: src/lkqd-videojs.js [hunk 7]
```diff
@@ -1,12 +1,13 @@
 var _merge = require("lodash/merge");
+var iframeId = new Date().getTime().toString() + Math.round(Math.random() * 1000000000).toString();
```
[the remark this answers is not in the document]
Comment (lpender) [reply src/lkqd-videojs.js]: haha

File: src/lkqd-videojs.js [hunk 8]
```diff
@@ -86,22 +81,59 @@ var LkqdVideoJS = (function() {
       this.options.creativeData,
       {
         slot: document.getElementById(this.options.containerEl.id),
-        videoSlot: document.getElementById(this.options.videoEl.id),
-        videoSlotCanAutoPlay: true
+        videoSlot: this.options.videoEl,
+        videoSlotCanAutoPlay: false,
```
[the remark this answers is not in the document]
Comment (lpender) [reply src/lkqd-videojs.js]: Testing will reveal if this is an issue--I suspect it's not.

**Pull request merged**